### PR TITLE
Data dictionaries post from Jia and myself

### PR DESCRIPTION
**Pull request by mikekaminsky**
cc @jialex

### CODE REVIEW

File: content/post/data_dictionaries.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+2. For compliance / security reasons, you have complex restrictions on who can access different parts of your warehouse
+
+These two use-cases are pretty niche, and we do not believe they apply to most internal-facing analytics teams. The second use-case is a bit tricky because what you want to build is not a “data dictionary” as we defined above (with “helpful” metadata about tables and columns) but rather an access-spec which can hierarchically define access to different parts of a warehouse (at the schema, table, column, and row level). We have some ideas on how you might build an effective tool for this, so if you’re working on something like that, please reach out!
+
+# Conclusion
+
+Overall, we think that data dictionaries tend not to be very useful, and a data team’s time will be better spent maintaining a well-organized code-base that is self-documenting and providing well-designed BI tools to end-users that do not require much documentation. Where possible, teams should make of tools for automatically generating documentation directly from code in order to prevent concept drift.
```
Comment (ilanman) [inline content/post/data_dictionaries.md]: `...teams should make use of tools...`

File: content/post/data_dictionaries.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+And everything is great...until you run into the issue of different audiences:
+
+* **Non-technical business users**: Just want the data, or at most, a high-level summary of the KPIs and why they’re important.
+* **Business analysts**: Want the data, but also the business logic behind it, and probably the data process used to populate the source and/or the relevant context.
+* **Data scientists / Engineers**: Just want the column-level metadata, relationships, and instructions to query the raw data correctly -- “I can write my own SQL/Python, thanks.”
+
+Which means that realistically, to solve everyone’s use case, you’ll need to write three versions of documentation (or one, very large version) for any given piece of data -- not fun.
+
+But even if you’re willing to go down that path...eventually something will change. Maybe your definition of NPS or Conversion just shifted a little bit (as we all know, business logic is a fickle master), and all of a sudden you have to update it not only in your data model and dashboards/reports, but also 3 documents and 2 diagrams.
```
[the remark this answers is not in the document]
Comment (mikekaminsky) [reply content/post/data_dictionaries.md]: Now that you've called it out, both of these phrases feel either gendered or problematic. Will revise!

**Comment by mikekaminsky** [comment]
@sbreitenother great points here! I think addressing all of these will make the post stronger.

* Airflow: yes, worth mentioning.
* Tool tips vs data dictionary: crucially, the tooltips exist _**in the BI tool**_  and are about the already-modeled data that the business user will use. Rather than being a list of columns that may not be applicable to end-users, it's very specifically _**only**_ about columns the end-user will want to use
* Good call on Carl's thing. We should link to that. I like it.
* Business analysts: this is tough -- were your business analysts querying the raw data, or just using looker? I think for this use case the wiki with a who-to-talk-to is probably the best bet. If you have people who are every week trying to model different aspects of the business, they're likely going to need to talk to an expert in those domains to keep from screwing something up
* Knowledge repo: good call out, and we should address this. I haven't seen it (though building something for this is next on my list of startup ideas). 

@jialex any thoughts?

**Comment by sbreitenother** [comment]
@mikekaminsky the Business Analysts couldn't write SQL, they could only use Looker. I like the who to talk to idea. My only challenge to that is it is a bit of a pain to keep that up as folks leave the team (especially because this happens in waves). I remember having to keep changing the name of who to talk to until we finally removed that section.

Re knowledge repo, the barriers I have seen are 1) dev resources, 2) authentication / securing the site and 3) the overhead of having one more internal knowledge base. At some point I researched ways to convert Jupyter notebooks into Confluence pages. I found an addon, but never tried it out.

**Comment by jialex** [comment]
@sbreitenother Definitely great feedback!

The way that I've approached empowering Business Analysts is a three step process:

1. Only BI team can create dashboards (so that every dashboard is guaranteed to be high quality, well-designed, very user friendly, and easy to understand)
2. We highly recommend exploring from dashboards, so even if you don't have all of the context, it's fairly simple to pick up the key dimensions/measures to use in a given scenario
3. We provide constant support through the Data Champions (power users who we collaborate closely with, and therefore have a good understanding), our slack channel, and our concept documentation (basically, high-level docs on a given metric which are linked to the dashboard in our dashboard directory)

I'm with you on the 'who to talk to' point -- generally speaking, I like to keep all of the critical context in the concept documentation and keep it up to date.  Tribal knowledge is fantastic, but never super scalable haha

**Pull request merged**